common/src/main/java/com/wynntils/models/wynnalphabet/WynnAlphabetModel.java

```diff
@@ -0,0 +1,368 @@
+
+            transcripted = numMatcher.replaceAll(match -> {
+                int numToTranscript = Integer.parseInt(match.group());
+
+                if (numToTranscript > MAX_TRANSCRIBABLE_NUMBER) {
+                    return "∞";
+                } else {
+                    return Models.WynnAlphabet.intToWynnicNum(numToTranscript);
+                }
+            });
+
+            for (char character : original.toCharArray()) {
+                Character replacement = Models.WynnAlphabet.transcribeEnglishToWynnic(character);
+
+                if (!replacement.equals(character)) {
+                    transcripted = transcripted.replace(character, replacement);
+                }
+            }
+        }
+
+        return transcripted;
+    }
+
+    public int wynnicNumToInt(String wynnicNum) {
+        int result = 0;
+
+        for (char num : wynnicNum.toCharArray()) {
+            int numIndex = wynnicNumbers.indexOf(num);
+
+            result += englishNumbers.get(numIndex);
+        }
+
+        return result;
+    }
+
+    public String intToWynnicNum(int number) {
+        StringBuilder wynnicNums = new StringBuilder();
+
+        int hundereds = number / 100;
+
+        number -= (hundereds * 100);
+
+        int fifties = number >= 50 ? 1 : 0;
+
+        number -= (fifties * 50);
+
+        int tens = number / 10;
+
+        number -= (tens * 10);
+
+        wynnicNums.append(String.valueOf(getOneHundered()).repeat(Math.max(0, hundereds)));
+        wynnicNums.append(String.valueOf(getFifty()).repeat(Math.max(0, fifties)));
+        wynnicNums.append(String.valueOf(getTen()).repeat(Math.max(0, tens)));
+
+        if (number > 0) {
+            wynnicNums.append(wynnicNumbers.get(englishNumbers.indexOf(number)));
+        }
+
+        return wynnicNums.toString();
+    }
+
+    public boolean shouldTranscribe(TranscribeMessagesFeature.TranscribeCondition condition, WynnAlphabet alphabet) {
+        return switch (condition) {
+            case NEVER -> false;
+            case TRANSCRIBER -> alphabet == WynnAlphabet.WYNNIC
+                    ? hasTranscriber(WynnAlphabet.WYNNIC)
+                    : hasTranscriber(WynnAlphabet.GAVELLIAN);
+            case DISCOVERY -> alphabet == WynnAlphabet.WYNNIC
+                    ? hasCompletedDiscovery(WynnAlphabet.WYNNIC)
+                    : hasCompletedDiscovery(WynnAlphabet.GAVELLIAN);
+            default -> true;
+        };
+    }
+
+    public boolean hasWynnicOrGavellian(String message) {
+        for (int i = message.length() - 1; i >= 0; i--) {
+            char c = message.charAt(i);
+
+            if (gavellianCharacters.contains(c) || wynnicCharacters.contains(c) || wynnicNumbers.contains(c)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public boolean containsBrackets(String message) {
+        return (message.contains("[[") && message.contains("]]")) || (message.contains("<<") && message.contains(">>"));
```

Comment:
To add a bit more layer of certainty that this matches correctly (and possibly clarity to the code), you can replace this with an regex, like `(\[\[.*\]\])|(<<.*>>)` or something like that.